### ISSUE TITLE
<Select> Not working

### ISSUE THREAD

**Issue by martindevans** [opitzconsulting/jquery-mobile-angular-adapter#75]
I have an angular controller setup like this:

```
function HostServerControl($scope) {
    $scope.Server = { name: "New Game", networked: true, lan: true, metaserver: true, gamemode: "construct" };

    $scope.Gamemodes = ["deathmatch", "construct"];
}
```

And that's bound to a selection box like this:

```
<select ng-model="Server.gamemode" ng-options="gamemode for gamemode in Gamemodes">
</select>
```

Clicking on this the first time causes this error and transitions me back to the previous screen:

```
jQuery.mobile-1.2.0.js Uncaught TypeError @ Line 7132: Cannot call method 'one' of undefined
```

Subsequently clicking again has no effect

**Comment by tbosch** [comment]
Hi,
could you create a jsfiddle for this? As far as I can see this seems to be correct...

Tobias

**Comment by martindevans** [comment]
While attempting to replicate this in a fiddle I found that the problem seems to have been caused by this:

<script>
    $(document).bind('mobileinit', function () {
        $.mobile.selectmenu.prototype.options.nativeMenu = false;
    });
</script>

This is referenced in the jQuery Mobile documentation here: http://jquerymobile.com/demos/1.0a4.1/docs/forms/forms-selects.html

N.b. I can't figure out how to include this in a fiddle, since I need to run that code between loading jQuery and jQuery mobile, which jsFiddle doesn't seem to allow. If you'd like I can host an example on my own server?

**Comment by tbosch** [comment]
You could use http://plnkr.co. There you can define the whole html page on your own. It's also good if you want to have an example with multiple pages that are loaded via ajax...

So, the example would be great!

Tobias

**Comment by martindevans** [comment]
I went ahead and hosted a demo on my own server instead, it seemed simpler. I've tried to cut out any unnecessary parts from the demo. Pay attention to the head of this page, it has the custom mobileinit I mentioned above included. Here it is:

http://www.placeholder-software.co.uk/static-files/.UI/Screens/MainMenu/MainMenu.html
1. You start off on the main menu, click new game
2. Click on any dropdown box
3. You should encounter an uncaught type error on line 7132 of jquery.mobile-1.2.0.js (Cannot call method 'one' of undefined)

I encounter this error with: Chrome, Awesomium, Internet Explorer 10

For reference, here is exactly the same thing without the mobileinit function, as you can see everything works perfectly:

http://www.placeholder-software.co.uk/static-files/.UI/Screens/MainMenu/MainMenu2.html
1. Click new game
2. Click a dropdown box and select an option
3. Options on dropdown 2 change to reflect valid options

**Comment by blasky** [comment]
I've been experiencing the same error when I attempt to use the "custom" mobile select menus with the adapter. The data.options.container that mobile expects is undefined at several key moments.

**Comment by blasky** [comment]
I was able to create a Fiddle with the necessary hooks at http://jsfiddle.net/ZHKBA/68/ (...can't be viewed in IE at all as the adapter script is blocked as having the wrong mime type...but Chrome will continue with just a warning).

This will throw 2 errors once you start interacting with the select control, one of which is the method "one" of undefined at jquery.mobile-1.2.0.js:7068 and the other is a method "unbind" of undefined at jquery.mobile-1.2.0.js:7001. Both of these errors are referring to a "container" element that is undefined at run time.

Note that if you comment out the following line everything executes without error:

```
 $.mobile.selectmenu.prototype.options.nativeMenu = false;
```

However, it is this line that is necessary to make use of mobile's custom menus for multi-select or custom styling or etc.

Thanks. (Very much enjoying the adapter so far. Just digging in more and more.)

**Comment by blasky** [comment]
Looks like some of the other routing and navigational changes that were made recently have resolved the null container issue here. I am no longer seeing the errors when I visit the fiddle above with the latest snapshot code. Cool.

However, the navigational steps for a large list still appear a little off. The fiddle below is the same as the previous except that the list of possible choices is too large to display inline as a popup. So mobile automatically displays the list in its select box dialog container...as a separate "page".

http://jsfiddle.net/ZHKBA/70/

After the fiddle loads, pull up the select list and select an option. Upon select, the browser will incorrectly navigate back off the page entirely or do absolutely nothing if there is no previous page to navigate to. Either way, the option selection event appears to be triggering a close operation on the select dialog and an unwanted back event of some kind.

(Note: If you pull up the select list and close it with the close button in the upper left corner without selecting an option the buggy behavior does not occur. There's some other code path in that case.)

**Comment by tbosch** [comment]
Hi,
could you check again, this should work now...

Tobias

**Comment by martindevans** [comment]
I updated to jquery-mobile-angular-adapter-1.1.2-SNAPSHOT.js, and it looks like this now works. Thanks very much :)

Unfortunately checkboxes broke when I updated but I'll open another issue for that

**Comment by blasky** [comment]
Awesome. Thanks so much.

The fiddle looks great. And my local code is now working for option selection as well.

(My local code no longer closes the dialog when I click on the x in the upper left corner, but that's working in the fiddle so I may have something in my html that is off. I noticed previously that mobile seemed to be very picky about how you structure your labels and placeholders for a select. I'll investigate.)

Thanks again.

**Comment by tbosch** [comment]
Closing this. Please open another ticket if you have further trouble with selects.

Tobias
